### ISSUE TITLE
Unused array not detected after array assignment

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1899]
```php
function foo() : void {
  $a = [];
  $a[0] = 1;
}
```
Expected: array is unused
Actual: no issue

https://psalm.dev/r/7d47ba378c
